google-cloud-clients/google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -2081,7 +2083,7 @@ private void retentionPolicyLockRequesterPays(boolean requesterPays) throws Exec
     }
     Bucket remoteBucket = storage.create(bucketInfo);
     try {
-      assertFalse(remoteBucket.retentionPolicyIsLocked());
+      assertNull(remoteBucket.retentionPolicyIsLocked());
```

Comment:
Long story short: https://github.com/GoogleCloudPlatform/google-cloud-java/blob/6235c62c63d00d9243f308bd34c3924fe56aa182/google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java#L878

Reply:
autogen?

Reply:
No autogen, still manual. The library had existing behavior of handling how fields are selected during a GET request. For example, if I don't select the field `Storage.BucketField#RETENTION_POLICY` then it's not returned in the bucket metadata resulting in `null`. If I do, then if a retention policy is not set on a bucket then it will be implicitly false or "null". If the field is selected and the field is returned then it's `true`. This method will never return false neither will existing Boolean method for versioning.

Reply:
Sorry, I should have read the entirety of the javadoc. Could you expand on why `com.google.cloud.storage.Storage.BucketField#RETENTION_POLICY` can be set and still return null? I would expect a `False` in that scenario.

Reply:
The GCS API doesn't return a value if the field is either `null` or `false`.  Only field that does work as expected is requester pays which is considered a bug due to inconsistency.

Reply:
Additionally the client library does not handle this behavior gracefully for existing Booleans.

Reply:
Ah, that's frustrating. I think using null as false is generally considered an anti-pattern, but I'm not sure in this case since the information is sometimes unknown.  From a usability perspective I feel like I would rather have `false` than `null` so I wouldn't be forced to do a null check. But probably outside the scope of this PR.

Reply:
It is outside scope, but it's also making an already bad issue worse by continuing the same behavior. I'll follow-up with a few people internally on this front. There's a better solution and I can do it now, but I'd need to resolve open questions with `Bucket.versioningEnable()` behavior changes first.

What are your thoughts on return a `Long` instead of a `Date` object?

Reply:
Long instead of a Date object for what?

Reply:
getRetentionPeriod?

Reply:
https://github.com/GoogleCloudPlatform/google-cloud-java/pull/3709/files#diff-2ca1569606e4f281912a4e4f5af3a408R1160

Reply:
DateTime from Apiary are represented as Long's